pkg/sql/catalog/dbdesc/database_desc.go

```diff
@@ -180,10 +179,28 @@ func (desc *Immutable) ForEachSchemaInfo(
 	return nil
 }
 
+// GetSchemaID returns the ID in the schema mapping entry for the
+// given name, 0 otherwise.
+func (desc *immutable) GetSchemaID(name string) descpb.ID {
+	info := desc.Schemas[name]
+	return info.ID
+}
+
+// GetNonDroppedSchemaName returns the name in the schema mapping entry for the
+// given ID, if it's not marked as dropped, empty string otherwise.
+func (desc *immutable) GetNonDroppedSchemaName(schemaID descpb.ID) string {
+	for name, info := range desc.Schemas {
+		if !info.Dropped && info.ID == schemaID {
+			return name
+		}
+	}
+	return ""
+}
+
```

Comment:
These two new methods replace `LookupSchema`, I feel this is a better way to expose the contents of the `Schemas` map which is otherwise quite easy to misinterpret.